configure.ac

```diff
@@ -184,13 +184,27 @@ AC_ARG_ENABLE([zstd],
               [enable_zstd="$enableval"],
               [enable_zstd=auto])
 
+have_zstd="no"
 AS_IF([test "x$enable_zstd" != "xno"], [
-  PKG_CHECK_MODULES([ZSTD], [libzstd], [have_zstd=yes], [have_zstd=no])
+  AC_CHECK_HEADERS([zstd.h],[
+    AC_PREPROC_IFELSE([
+      AC_LANG_SOURCE([
+  #include <zstd.h>
+  #if ((ZSTD_VERSION_MAJOR < 1) || (ZSTD_VERSION_MAJOR == 1 && ZSTD_VERSION_MINOR < 3) || (ZSTD_VERSION_MAJOR == 1 && ZSTD_VERSION_MINOR == 3 && ZSTD_VERSION_RELEASE < 8))
+  #error ZSTD too old
+  #endif
+      ])
+    ],[ have_zstd="yes" ])
+  ])
+  AC_MSG_CHECKING(for zstd >= 1.3.8)
+  AC_MSG_RESULT($have_zstd)
+
   AS_IF([test "$enable_zstd" = "yes"], [
     if test "$have_zstd" = "no"; then
       AC_MSG_ERROR([--enable-zstd specified, but not available])
     fi
   ])
+  PKG_CHECK_MODULES([ZSTD], [libzstd], [have_zstd=yes], [have_zstd=no])
```

Comment:
I think it is much easier to have here `libzstd >= 1.3.8` rather than custom C preproc check.

Reply:
Oh yeah, it is new to me! Thanks.